### ISSUE TITLE
Add a README giving description of each example for convenience.

### ISSUE THREAD

**Issue by shipsywor** [superdurszlak/bad-concurrency-examples#1]
It would be helpful to explain and give a description of each example right in the README.

**Comment by superdurszlak** [comment]
Agree, but bear in mind that this repo has barely been started and benchmarks are still WIP.

**Comment by shipsywor** [comment]
Thank you! Looking forward to it. This could be a great repo.

**Comment by superdurszlak** [comment]
Unfortunately, I did not have much time to get back to this project and hone it, and I cannot promise to improve it anytime soon. 

There is a lot of areas that are begging for improvement before it becomes relatable to anything:
* Specific benchmarks are not documented at all
* I basically added scenarios ad-hoc since I wanted to show a couple of basic concepts at my presentation elsewhere - they should be more thought out, but I rushed to get them done in no time
* My familiarity with JMH is questionable at best, and it shows
* The sheer number of variables makes it a great example of the curse of dimensionality, and thus corners had to be cut:
  * matrix/vector sizes - necessary to show various effects
  * block sizes in several dimensions, where parallelization or optimization for cache required subdividing the task
  * sequential/parallel execution
  * several approaches at concurrent execution - with bare threads, thread pool, one could probably think of several others as well
  * row-major and column-major matrix alignment in memory, and I did not even bother to consider sparse matrices (the point was to show compute-intensive, memory-intensive and IO-intensive operations, not optimizing matrix operations per se)
  * several approaches to thread safety of matrix operations (even worse, they are not even fully thread-safe)
  * execution with and without cache optimization (which is sub-optimal anyway, but since I wrote it in a hurry, I did not attempt to utilize cache to its fullest)

**Comment by shipsywor** [comment]
Hey, whenever you get time, keep the project alive. Thanks for the effort and details.
